news/tasks.py

```diff
@@ -52,18 +58,23 @@
     'news.tasks.add_sms_user',
     'news.tasks.add_sms_user_optin',
 ]
+if hasattr(cache, 'lock'):
```

Comment:
it has that if the cache is redis. [`django-redis` exposes the lock method from the python redis lib](http://niwinz.github.io/django-redis/latest/#_locks).